pkg/reconciler/configuration/configuration.go

```diff
@@ -167,38 +167,36 @@ func (c *Reconciler) reconcile(ctx context.Context, config *v1alpha1.Configurati
 	switch {
 	case rc == nil || rc.Status == corev1.ConditionUnknown:
 		logger.Infof("Revision %q of configuration is not ready", revName)
-		if err = c.findAndSetLatestReadyRevision(config); err != nil {
-			return fmt.Errorf("failed to find and set latest ready revision: %w", err)
-		}
 
 	case rc.Status == corev1.ConditionTrue:
 		logger.Infof("Revision %q of configuration is ready", revName)
-
-		created, ready := config.Status.LatestCreatedRevisionName, config.Status.LatestReadyRevisionName
-		if ready == "" {
+		if config.Status.LatestReadyRevisionName == "" {
 			// Surface an event for the first revision becoming ready.
 			c.Recorder.Event(config, corev1.EventTypeNormal, "ConfigurationReady",
 				"Configuration becomes ready")
 		}
-		// Update the LatestReadyRevisionName and surface an event for the transition.
-		config.Status.SetLatestReadyRevisionName(lcr.Name)
-		if created != ready {
-			c.Recorder.Eventf(config, corev1.EventTypeNormal, "LatestReadyUpdate",
-				"LatestReadyRevisionName updated to %q", lcr.Name)
-		}
 
 	case rc.Status == corev1.ConditionFalse:
 		logger.Infof("Revision %q of configuration has failed", revName)
-
 		// TODO(mattmoor): Only emit the event the first time we see this.
 		config.Status.MarkLatestCreatedFailed(lcr.Name, rc.Message)
 		c.Recorder.Eventf(config, corev1.EventTypeWarning, "LatestCreatedFailed",
 			"Latest created revision %q has failed", lcr.Name)
+
+	default:
+		return fmt.Errorf("unrecognized condition status: %v on revision %q", rc.Status, revName)
+	}
+	if rc != nil && rc.Status == corev1.ConditionTrue {
+		old, new := config.Status.LatestReadyRevisionName, lcr.Name
+		config.Status.SetLatestReadyRevisionName(lcr.Name)
+		if old != new {
+			c.Recorder.Eventf(config, corev1.EventTypeNormal, "LatestReadyUpdate",
+				"LatestReadyRevisionName updated to %q", new)
+		}
+	} else {
```

Comment:
Actually it won't work for the byo revision name rollback use case. Because `findAndSet...` will try to get the list of created revision starting from the generation number of the current LRR, it won't fetch the rollback revision and consequently update LRR to the correct previous revision.